src/Microsoft.SqlTools.ServiceLayer/DacFx/ExtractOperation.cs

```diff
@@ -28,7 +29,14 @@ public ExtractOperation(ExtractParams parameters, ConnectionInfo connInfo) : bas
 
         public override void Execute()
         {
-            this.DacServices.Extract(this.Parameters.PackageFilePath, this.Parameters.DatabaseName, this.Parameters.ApplicationName, this.Parameters.ApplicationVersion, null, null, null, this.CancellationToken);
+            // try to get version
+            Version version;
+            Version.TryParse(this.Parameters.ApplicationVersion, out version);
```

Comment:
Version is value type, so won't comparing to null always be false? Version.TryParse() returns a boolean, so you can put that method invocation as if statement conditional expression: `if (!TryParse(...)) throw`

Reply:
The comparison was true when the version was invalid, but I'll change it since it's more clear to have the Version.TryParse() in the if statement.

Reply:
Turns out version is not a value type  :)